### PR TITLE
test produced SDOs

### PR DESCRIPTION
**Pull request by jo-pol**
@DANS-KNAW/easy please review

### CODE REVIEW

File: src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala [hunk 1]
```diff
@@ -48,4 +50,24 @@ class EasyStageFileItemSpec extends FlatSpec with Matchers {
     val args = "src/test/resources/example.csv outdir".split(" ")
     getSettingsRows(new FileItemConf(args)).get.size shouldBe 5
   }
+
```
Comment (janvanmansum) [inline src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala:57]: This test really needs to be more specific. Just checking that the entire contents of a file is as a stored example is bound to give false positives (i.e. tests that fail although nothing is broken, just the output changed in some trivial way). The title of the test is also a clear indication that this is not testing something clearly defined.

There are many invariants that can be tested here. To name just a few:
- Of all the files in the list: that they must exist after the test
- that the id of the folder appears where expected in these files
- that the size appears where expected

Better even would be to test lower level functions (`createFileSdo`, `createFolderSdo`). Of course, that doesn't mean that below test cannot be done as well.

**Comment by janvanmansum** [comment]
:-1: Not happy about the test, yet. Please, review my comments.

File: src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala (hunk 1 above)
Comment (jo-pol) [inline src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala:57]: The failing travis build while the local build succeeds proofs your point. I'm currently considering extracting leaf elements of the expected output and checking their presence in the actual output.

File: src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala [hunk 2]
```diff
@@ -48,4 +60,138 @@ class EasyStageFileItemSpec extends FlatSpec with Matchers {
...
+        // TODO fix irrelevant differences for fo.xml
+        new File(actual) should haveSameContentAs(new File(expected))
+      case (actual, expected) =>
+        // metadata of a file or folder
+        readFlatXml(actual) shouldBe readFlatXml(expected)
+    }
+  }
+
+  def readFlatXml(file: String): Seq[Node] =
+    XML.loadFile(file).descendant.toSeq.filter(n => n.toString().startsWith("<")).sortBy(n => n.label)
```
Comment (jo-pol) [inline src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala]: @DANS-KNAW/easy has anyone a better idea for the filter?

File: src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala [hunk 3]
```diff
@@ -48,4 +54,125 @@ class EasyStageFileItemSpec extends FlatSpec with Matchers {
...
+    ).foreach {
+      case (actual, expected) if actual.endsWith("cfg.json") =>
+        readCfgJson(expected) shouldBe readCfgJson(actual)
+      case (actual, expected) if actual.endsWith("fo.xml") =>
+        readDatastreamFoxml(actual) shouldBe readDatastreamFoxml(expected)
+      case (actual, expected) => // metadata of a file or folder
+        readFlatXml(actual) shouldBe readFlatXml(expected)
+    }
+
+    // a less verbose check reworded
```
Comment (jo-pol) [inline src/test/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItemSpec.scala:98]: @DANS-KNAW/easy 
The comment by Jan above applies to an earlier commit that compared the exact contents of the actual and expected files. I understand Jan wants to see checks as shown below (currently incomplete as it compares the content of just one file) but IMHO the expected output under the test resources makes it easier to picture what is produced and thus better documents the application. The current implementation is no longer sensitive for irrelevant changes such as white space and order of elements or attributes. The methods in [SdoFiles](https://github.com/DANS-KNAW/easy-stage-dataset/pull/62/files#diff-19a676551f79b0048cb9c643b8eded83R32) take care of that.

**Comment by jo-pol** [comment]
integrated in #64

**Pull request closed without merging**